packages/formik/src/Formik.tsx

```diff
@@ -5,29 +5,48 @@ import isPlainObject from 'lodash/isPlainObject';
 import {
   FormikConfig,
   FormikErrors,
-  FormikState,
+  FormikReducerState,
   FormikTouched,
   FormikValues,
   FormikProps,
   FieldMetaProps,
   FieldHelperProps,
   FieldInputProps,
   FormikHelpers,
-  FormikHandlers,
+  FormikApi,
+  HandleBlurFn,
+  HandleBlurEventFn,
+  HandleChangeEventFn,
+  HandleChangeFn,
+  FormikState,
 } from './types';
 import {
   isFunction,
   isString,
   setIn,
   isEmptyChildren,
   isPromise,
-  setNestedObjectValues,
   getActiveElement,
   getIn,
   isObject,
+  setNestedObjectValues,
 } from './utils';
 import { FormikProvider } from './FormikContext';
 import invariant from 'tiny-warning';
+import { useIsomorphicLayoutEffect } from './hooks/useIsomorphicLayoutEffect';
+import {
+  Comparer,
+  Selector,
+  useOptimizedSelector,
+} from './hooks/useOptimizedSelector';
+import { unstable_batchedUpdates } from 'react-dom';
```

Comment:
how does this work with react native?

Reply:
I'll have to test it out, but I think you're right this needs a different import for react native. I've never actually used React Native, but maybe we should have an expo version of app/ for testing?!

Reply:
We don't need to manually batch updates here anymore as I moved this into an effect. I'll remove and test.